### PR TITLE
=str #22448 rewrite Source.empty to GraphStage

### PR DESCRIPTION
**Pull request by ktoso**
Solves https://github.com/akka/akka/issues/22448

Deserves backport

### CODE REVIEW

File: akka-docs/rst/scala/code/docs/actor/LOLSPEC.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      }
+    }))
+
+    system.eventStream.subscribe(miku, classOf[Object])
+
+    system.eventStream.publish("Hello!")
+  }
+
+}
+
```
Comment (johanandren) [inline akka-docs/rst/scala/code/docs/actor/LOLSPEC.scala]: Wat? O.o

**Review by johanandren — APPROVED** [review]
LGTM (except the weird lolspec)

File: akka-docs/rst/scala/code/docs/actor/LOLSPEC.scala (hunk 1 above)
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/actor/LOLSPEC.scala]: lol, yeah i was sanity checking something for a support ticket :D Thanks for noticing...

File: akka-stream/src/main/scala/akka/stream/impl/Sources.scala [hunk 2]
```diff
@@ -396,3 +398,22 @@ final class LazySource[T, M](sourceFactory: () ⇒ Source[T, M]) extends GraphSt
...
+/** INTERNAL API */
+@InternalApi
+final object EmptySource extends GraphStage[SourceShape[Nothing]] {
+  val out = Outlet[Nothing]("EmptySource.out")
+  override val shape = SourceShape(out)
+
+  override protected def initialAttributes = DefaultAttributes.lazySource
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with OutHandler {
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/Sources.scala]: could we even return the same `GraphStageLogic` instance every time here?

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/Sources.scala]: No, it has mutable state

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/Sources.scala]: I mean internally

**Review by drewhk — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/impl/Sources.scala (hunk 2 above)
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/Sources.scala]: ok

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by ktoso** [comment]
Will backport

**Pull request merged**